### ISSUE TITLE
The backpropagation part is needed?

### ISSUE THREAD

**Issue by xuqingyu26** [nicodjimenez/lstm#50]
I think with the definition of the network and loss function, the backpropagation should be auto computed, why should we explicitly definite it in the code?

**Comment by RNGrunshen** [comment]
> I think with the definition of the network and loss function, the backpropagation should be auto computed, why should we explicitly definite it in the code?

because we are using numpy instead of torch, it is necessary for us to build the calculation diagram manually
